lib/releaselib.sh

```diff
@@ -1099,6 +1101,91 @@ release::docker::release () {
   return 0
 }
 
+# TODO(vdf): Consider collapsing this into release::docker::release and renaming
+#            that function AFTER the k8s.gcr.io Vanity Domain Flip (VDF).
+###############################################################################
+# Validates that image manifests have been pushed to a specified remote registry.
+# Uses 'skopeo inspect'.
+#
+# @param registry - docker registry
+# @param version - version tag
+# @param build_output - build output directory
+# @return 1 on failure
+release::docker::validate_remote_manifests () {
+  local registry=$1
+  local push_registry=$registry
+  local version=$2
+  local build_output=$3
+  local release_images=$build_output/release-images
+  local arch
+  local -a arches
+  local tarfile
+  local orig_tag
+  local new_tag
+  local binary
+  local -A manifest_images
+
+  # TODO(vdf): Remove all GCRIO_PATH_PROD_PUSH logic once the k8s.gcr.io vanity
+  #            domain flip (VDF) is successful
+  if [[ "$registry" == "$GCRIO_PATH_PROD" ]]; then
+    # Switch to the push alias if using the $GCRIO_PATH_PROD alias
+    push_registry="$GCRIO_PATH_PROD_PUSH"
+  fi
+
+  logecho "Validating image manifests in $push_registry..."
+
+  arches=($(ls "$release_images"))
+  for arch in "${arches[@]}"; do
+    for tarfile in $release_images/$arch/*.tar; do
+      # There may be multiple tags; just get the first
+      orig_tag=$(tar xf $tarfile manifest.json -O  | jq -r '.[0].RepoTags[0]')
+      if [[ ! "$orig_tag" =~ ^.+/(.+):.+$ ]]; then
+        logecho "$FAILED: malformed tag in $tarfile:"
+        logecho $orig_tag
+        return 1
+      fi
+      binary=${BASH_REMATCH[1]}
+
+      new_tag="$push_registry/${binary/-$arch/}"
+      new_tag_with_arch=("$new_tag-$arch:$version")
+      manifest_images["${new_tag}"]+=" $arch"
+    done
+  done
+
+  for image in "${!manifest_images[@]}"; do
+    local archs
+    local manifest
+    local digest
+
+    logecho "Validating manifest list exists for ${image}:${version}..."
+
+    archs=$(echo "${manifest_images[$image]}" | sed -e 's/^[[:space:]]*//')
+
+    if ! skopeo inspect "docker://${image}:${version}" --raw >/dev/null 2>&1; then
+      logecho "Could not find manifest list for ${image}:${version}"
+      return 1
+    else
+      manifest=$(skopeo inspect "docker://${image}:${version}" --raw)
+    fi
```

Comment:
Have you tried:
```bash
    manifest=$(skopeo inspect "docker://${image}:${version}" --raw >/dev/null 2>&1)
    if [ $? ]; then
      logecho "Could not find manifest list for ${image}:${version}"
      return 1
    fi
```

Reply:
I opted for a bit of a combination of the two:

```shell
    if ! manifest=$(skopeo inspect "docker://${image}:${version}" --raw); then
      logecho "Could not find manifest list for ${image}:${version}"
      return 1
    fi
```

See the guidance on shellcheck SC2181: https://github.com/koalaman/shellcheck/wiki/SC2181